packages/gui/src/components/ModeSelector.tsx

```diff
@@ -39,7 +37,7 @@ const ModeSelector = () => {
   const [open, setOpen] = useState(false)
   const [disabled, setDisabled] = useState(false)
   const [showModeDialog, setShowModeDialog] = useState(false)
-  const { modeName, _setModeName, prevMode, modes } = useContext(ModeContext)
```

Comment:
Do we need `prevMode` at this point? We might can remove it from the context as well as a state property.